File: main_module/src/main/java/com/mobile/main_module/login/model/LoginModelImpl.java
package com.mobile.main_module.login.model;

import com.mobile.common_library.manager.RetrofitManager;
import com.mobile.main_module.login.request.LoginRequest;

import java.util.Map;

import io.reactivex.Observable;
import okhttp3.ResponseBody;

/**
 * author    : Urasaki
 * e-mail    : Urasaki@qq.com
 * date      : 2019/3/10 17:03
 * introduce :
 */

public class LoginModelImpl implements ILoginModel {
	
	private static final String TAG = "LoginModelImpl";
	
	public LoginModelImpl() {
	}
	
	@Override
	public Observable<ResponseBody> getAuthCode(Map<String, String> bodyParams) {
		return RetrofitManager.getInstance().getRetrofit()
				   .create(LoginRequest.class)
				   .getAuthCode(bodyParams);
	}
	
	@Override
	public Observable<ResponseBody> loginWithAuthCode(Map<String, String> bodyParams) {
		return RetrofitManager.getInstance().getRetrofit()
				   .create(LoginRequest.class)
				   .getLoginWithAuthCodeData(bodyParams);
	}
	
	@Override
	public Observable<ResponseBody> register(Map<String, String> bodyParams) {
		return RetrofitManager.getInstance().getRetrofit()
				   .create(LoginRequest.class)
				   .getRegisterData(bodyParams);
	}
	
}

File: module_main/src/main/java/com/phone/module_main/login/model/LoginModelImpl.kt
package com.phone.module_main.login.model

import com.phone.library_common.manager.RetrofitManager.Companion.get
import com.phone.module_main.login.request.LoginRequest
import io.reactivex.Observable
import okhttp3.ResponseBody

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2019/3/10 17:03
 * introduce :
 */

class LoginModelImpl : ILoginModel {

    private val TAG = LoginModelImpl::class.java.simpleName

    override fun getAuthCode(bodyParams: Map<String, String>): Observable<ResponseBody> {
        return get().getRetrofit()
            .create(LoginRequest::class.java)
            .getAuthCode(bodyParams)
    }

    override fun loginWithAuthCode(bodyParams: Map<String, String>): Observable<ResponseBody> {
        return get().getRetrofit()
            .create(LoginRequest::class.java)
            .getLoginWithAuthCodeData(bodyParams)
    }

    override fun register(bodyParams: Map<String, String>): Observable<ResponseBody> {
        return get().getRetrofit()
            .create(LoginRequest::class.java)
            .getRegisterData(bodyParams)
    }

}